extensions/positron-r/src/r-installation.ts

```diff
@@ -25,20 +25,82 @@ export interface RMetadataExtra {
 	 * https://github.com/posit-dev/positron/issues/2659
 	 */
 	readonly current: boolean;
+
+	/**
+	 * How did we discover this R binary?
+	 */
+	readonly reasonDiscovered: ReasonDiscovered[] | null;
+}
+
+/**
+ * Enum represents how we discovered an R binary.
+ */
+export enum ReasonDiscovered {
+	affiliated,
+	registry,
+	/* eslint-disable @typescript-eslint/naming-convention */
+	PATH,
+	HQ,
+	/* eslint-enable @typescript-eslint/naming-convention */
+	adHoc,
+	user
+}
+
+/**
+ * Enum represents why we rejected an R binary.
+ */
+export enum ReasonRejected {
+	invalid,
+	unsupported,
+	nonOrthogonal
+}
```

Comment:
Responding to an earlier request from @jmcphers:

> Idiomatically I think it'd be better to leave the enum with the default value representation (string or number); if you want to create a user-facing string that describes the enum value, have a function do it.

https://github.com/posit-dev/positron/pull/4878/files#r1785400981

Reply:
Here are some examples of how reasons appear in the log, when they're processed through `friendlyReason()`.

<details style=\"margin-bottom:10px;\">
<summary>Usable installation</summary>

```
2024-12-06 12:13:40.397 [info] R installation discovered: {
  "usable": true,
  "supported": true,
  "reasonDiscovered": [
    "Found in PATH, via the `which` command",
    "Found in the primary location for R versions on this operating system",
    "Found in a conventional location for symlinked R binaries"
  ],
  "reasonRejected": null,
  "binpath": "/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/bin/R",
  "homepath": "/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources",
  "semVersion": {
    "options": {},
    "loose": false,
    "includePrerelease": false,
    "raw": "4.4.2",
    "major": 4,
    "minor": 4,
    "patch": 2,
    "prerelease": [],
    "build": [],
    "version": "4.4.2"
  },
  "version": "4.4.2",
  "arch": "arm64",
  "current": true,
  "orthogonal": true
}
```

</details>

<details style=\"margin-bottom:10px;\">
<summary>Unusable installation</summary>

```
2024-12-06 12:13:40.397 [info] R installation discovered: {
  "usable": false,
  "supported": true,
  "reasonDiscovered": [
    "Found in the primary location for R versions on this operating system"
  ],
  "reasonRejected": "Non-orthogonal installation that is also not the current version",
  "binpath": "/Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/bin/R",
  "homepath": "/Library/Frameworks/R.framework/Resources",
  "semVersion": {
    "options": {},
    "loose": false,
    "includePrerelease": false,
    "raw": "4.4.2",
    "major": 4,
    "minor": 4,
    "patch": 2,
    "prerelease": [],
    "build": [],
    "version": "4.4.2"
  },
  "version": "4.4.2",
  "arch": "arm64",
  "current": false,
  "orthogonal": false
}
```

</details>